_field-types/index.md

```diff
@@ -14,38 +14,57 @@ redirect_from:
+If you're starting to build out your cluster and data, you may not know exactly how your data should be stored. In those cases, you can use dynamic mappings, which tell OpenSearch to dynamically add data and its fields. However, if you know exactly what types your data falls under and want to enforce that standard, then you can use explicit mappings.
 
 For example, if you want to indicate that `year` should be of type `text` instead of an `integer`, and `age` should be an `integer`, you can do so with explicit mappings. By using dynamic mapping, OpenSearch might interpret both `year` and `age` as integers.
 
-This section provides an example for how to create an index mapping and how to add a document to it that will get ip_range validated.
+This documentation provides an example for how to create an index mapping and how to add a document to it that will get `ip_range` validated.
 
-#### Table of contents
-1. TOC
-{:toc}
-
-
----
 ## Dynamic mapping
 
 When you index a document, OpenSearch adds fields automatically with dynamic mapping. You can also explicitly add fields to an index mapping.
 
-#### Dynamic mapping types
+### Dynamic mapping types
 
 Type | Description
 :--- | :---
 null | A `null` field can't be indexed or searched. When a field is set to null, OpenSearch behaves as if that field has no values.
-boolean | OpenSearch accepts `true` and `false` as boolean values. An empty string is equal to `false.`
+Boolean | OpenSearch accepts `true` and `false` as Boolean values. An empty string is equal to `false.`
 float | A single-precision 32-bit floating point number.
 double | A double-precision 64-bit floating point number.
 integer | A signed 32-bit number.
 object | Objects are standard JSON objects, which can have fields and mappings of their own. For example, a `movies` object can have additional properties such as `title`, `year`, and `director`.
-array | Arrays in OpenSearch can only store values of one type, such as an array of just integers or strings. Empty arrays are treated as though they are fields with no values.
+array | Arrays in OpenSearch can only store values of one type, such as an array of only integers or strings. Empty arrays are treated as though they are fields with no values.
 text | A string sequence of characters that represent full-text values.
 keyword | A string sequence of structured characters, such as an email address or ZIP code.
 date detection string | Enabled by default, if new string fields match a date's format, then the string is processed as a `date` field. For example, `date: "2012/03/11"` is processed as a date.
 numeric detection string | If disabled, OpenSearch may automatically process numeric values as strings when they should be processed as numbers. When enabled, OpenSearch can process strings into `long`, `integer`, `short`, `byte`, `double`, `float`, `half_float`, `scaled_float`, and `unsigned_long`. Default is disabled.
 
```

Comment:
Can we cover all field types from here? https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/indices/IndicesModule.java#L151-L174

Reply:
Should we link the reader to the GitHub file instead of listing the mapping types in the documentation?

Reply:
Should we link to the GitHub link instead of listing the field types in the documentation?

Reply:
Linking to code can be tricky as the code can move due to refactoring making the documentation point to old/unusable links. Hence, I'd prefer to keep it within the documentation if possible.